connect/src/state/context.rs

```diff
@@ -271,20 +279,21 @@ impl ConnectState {
 
     pub fn merge_context(&mut self, context: Option<Context>) -> Option<()> {
         let mut context = context?;
-        if self.context_uri() != &context.uri {
+        if matches!(context.uri, Some(ref uri) if uri != self.context_uri()) {
             return None;
         }
 
         let current_context = self.context.as_mut()?;
         let new_page = context.pages.pop()?;
 
         for new_track in new_page.tracks {
-            if new_track.uri.is_empty() {
+            if new_track.uri.is_none() || matches!(new_track.uri, Some(ref uri) if uri.is_empty()) {
```

Comment:
That's a thing :o? Fancy, will do later.

Reply:
MSRV upgrades usually generate extra work for library users, please upgrade sparingly. For syntax stuff I don't think it's worth the headaches.

Reply:
Ah I see. But yeah, I also wouldn't upgrade just because of syntax sugar (even tho my inner me really wants too). 

Anyhow we seem to have other dependencies that require newer MSRV versions (see #1425 for example). So that might be a different point of view. But as I can't see the consequences, It's all just a suggestions^^